### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by charlesbluca** [fugue-project/fugue#526]
**Minimal Code To Reproduce**

```python
import fugue_sql

dag = fugue_sql.FugueSQLWorkflow()
df = dag.df([[0, "hello"], [1, "world"]], "a:int64,b:str")
dag("SELECT * FROM df WHERE a > 0 YIELD DATAFRAME AS result")

result = dag.run("dask")
```

**Describe the bug**
When pulling in `triad>=0.9.2`, the above reproducer fails due to a missing `enforce_type` attribute:

```python-traceback
```

**Expected behavior**

**Environment (please complete the following information):**
 - Backend: pandas/dask/ray?
 - Backend version:
 - Python version:
 - OS: linux/windows

**Comment by goodwanghan** [comment]
Ah, sorry, but I think Fugue side has finished the change to depend on dask-sql when the backend is dask, I will make a pr to dask-sql to remove the unnecessary part.

**Comment by paolorechia** [comment]
Hey, I'm also affected by this bug, using Spark backend.

Some relevant requirements:
```txt
adtk==0.6.2
pyod==1.0.1
numpy==1.22.3
fugue==0.8.3
```

My application suddenly stopped working from one deployment to another. It seems like the [setup.py](https://github.com/fugue-project/fugue/blob/master/setup.py) of fugue pins down versions in a rather risky way:

```python
        "triad>=0.9.3",
        "adagio>=0.2.4",
        # sql dependencies
        "qpd>=0.4.4",
        "fugue-sql-antlr>=0.1.6",
```

So I'm guessing `triad` was updated, which caused my pinned down version of `fugue` to break. I see two long term solutions to this problem:

1. Change the pinning strategy of the fugue project, to something like this:
```python
        "triad<=0.9.3",
        "adagio<=0.2.4",
        # sql dependencies
        "qpd<=0.4.4",
        "fugue-sql-antlr<=0.1.6",
```
This would ensure sub-dependencies are only updated with a specific version update on fugue.

2. Add a disclaimer in the installation steps that proper requirement locking is required from end-users, by using something like `pipenv`. Currently documentation recommends installation by `pip` which is a ticking bomb: https://github.com/fugue-project/fugue#installation

**Comment by cmditch** [comment]
Agreed. Using only `>=` on a package where "patch" updates are liable to have major changes is poor form.
